doc/api/vm.md

```diff
@@ -126,6 +126,15 @@ Creates a code cache that can be used with the `Script` constructor's
 `cachedData` option. Returns a `Buffer`. This method may be called at any
 time and any number of times.
 
+The code cache of the `Script` doesn't contain any JavaScript observable
+states. The code cache is safe to be saved along side the script source and
+used to construct new `Script` instances multiple times.
+
+Functions in the `Script` source can be marked as lazily compiled and they are
+not compiled at construction of the `Script`. These functions are forced to be
+compiled when they are invoked the first time. The code cache serializes the
+current compilation state.
```

Comment:
```suggestion
metadata that V8 currently knows about the `Script` that it can use to speed up
future compilations.
```

The current wording might lead to misbeliefs that V8 could serialize optimized code.